### ISSUE TITLE
Why inference on custom dataset using sparseConv results into much larger pointclouds than input data?

### ISSUE THREAD

**Issue by anayghatpande** [isl-org/Open3D-ML#562]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D-ML/issues).
- [X] I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

Hello,
I use a custom dataset for bin-picking and training on sparseConvUnet with semantic segmentation. The training results are pretty good, as shown in the figures. 
![train_accuracy](https://user-images.githubusercontent.com/44940338/189702266-3ec7fbaa-b0dc-4f66-aca1-1225fd2a978c.png)
![train_loss](https://user-images.githubusercontent.com/44940338/189702346-3b5a86dd-b2fd-4c1d-bf5f-39aa2d57b35b.png)
![val_loss](https://user-images.githubusercontent.com/44940338/189702289-a37e50fb-1743-4b3a-a36f-09c3d9118552.png)

**Although the inference on the training data results was eight times bigger than the actual data labels, I tried a visualizer from open3d-ml and visualized it manually. In both cases, I am getting bad segmentation results.**
This is actual test data:
![test_img](https://user-images.githubusercontent.com/44940338/189703186-3958ac4b-59c0-4aaf-a173-55db479795f4.png)
Open3d-ml Visualiser:
![spars_conv](https://user-images.githubusercontent.com/44940338/189702889-d2517497-e4d9-4b88-9e7a-60d48422e9be.png)

Please find here the code for the data loading to inference: Visualisation.py
```
for j in range(len(pc_names)):
        train_split = dataset.get_split("train")
        pc_data0 = train_split.get_data(n[j])
        pc_data.append(pc_data0)

def pred_custom_data(pc_names, pcs, pipeline_k):
    vis_points = []
    for i, data in enumerate(pcs):
        name = pc_names[i]
        results_k = pipeline_k.run_inference(data)
        pred_label_k = (results_k['predict_labels']+1).astype(np.int32)
        label = data['label']
        pts = data['point']
        feature = data['feat']
        vis_d = {
            "name": name,
            "points": pts,
            "feat": feature,
            "labels": label,
            "pred": pred_label_k,
        }
        vis_points.append(vis_d)
     return vis_points

```
For complete code: [https://github.com/anayghatpande/Master_Thesis](url)
I think there is something wrong with the inference evaluator, as the training accuracy is quite good. Also, I can see that the predicted labels resemble the trained samples. But it does not appear to be predicted properly with the current inference script.
Please help me resolve this problem which I feel is either in the **data loader, the visualizer, or the inference evaluator.** 
Thanks.
